yaksh/views.py

```diff
@@ -1176,3 +1200,29 @@ def remove_teachers(request, course_id):
         teachers = User.objects.filter(id__in=teacher_ids)
         course.remove_teachers(*teachers)
     return my_redirect('/exam/manage/courses')
+
+
+def test_mode(user, godmode=False, questions_list=None, quiz_id=None):
+    """creates a trial question paper for the moderators"""
+
+    if questions_list is not None:
+        trial_course = Course.objects.create_trial_course(user)
+        trial_quiz = Quiz.objects.create_trial_quiz(trial_course, user)
+        trial_questionpaper = QuestionPaper.objects\
+                                           .create_trial_paper_to_test_questions\
+                                            (trial_quiz, questions_list)
+    else:
+        trial_quiz = Quiz.objects.create_trial_from_quiz(quiz_id, user, godmode)
+        trial_questionpaper = QuestionPaper.objects\
+                                           .create_trial_paper_to_test_quiz\
+                                            (trial_quiz, quiz_id)
+    return trial_questionpaper
+
+
+@login_required
+def test_quiz(request, mode, quiz_id):
+    """creates a trial quiz for the moderators"""
+    godmode = True if mode == "godmode" else False
+    current_user = request.user
+    trial_questionpaper = test_mode(current_user, godmode, None, quiz_id)
+    return my_redirect("/exam/start/{0}".format(trial_questionpaper.id))
```

Comment:
After testing questions or taking trial_quiz, moderator is redirected to his dashboard.
It will be good if you redirect to the same page where we click "test",  "user mode" or "god mode" buttons.